java-frontend/src/main/java/org/sonar/java/resolve/BytecodeVisitor.java

```diff
@@ -198,8 +199,12 @@ public FieldVisitor visitField(int flags, String name, String desc, @Nullable St
     Preconditions.checkNotNull(desc);
     if (isNotSynthetic(flags)) {
       //Flags from asm lib are defined in Opcodes class and map to flags defined in Flags class
-      final JavaSymbol.VariableJavaSymbol symbol = new JavaSymbol.VariableJavaSymbol(Flags.filterAccessBytecodeFlags(flags),
-          name, convertAsmType(org.objectweb.asm.Type.getType(desc)), classSymbol);
+      int filteredFlags = Flags.filterAccessBytecodeFlags(flags);
+      JavaType type = convertAsmType(Type.getType(desc));
+      JavaSymbol.VariableJavaSymbol symbol = new JavaSymbol.VariableJavaSymbol(filteredFlags, name, type, classSymbol);
```

Comment:
it should be dropped and GC anyway, but I would avoid creating a variable if it is of no use.

Reply:
Fixed in f7ea69796727befbad0ae1ec1e09982ec507b652